pulp_container/app/viewsets.py

```diff
@@ -114,7 +116,50 @@ class Meta:
         }
 
 
-class TagViewSet(ReadOnlyContentViewSet):
+Repo = namedtuple("Repo", "type perm")
+mirror_repo = Repo("container.container", "container.view_containerrepository")
+push_repo = Repo("container.container-push", "container.view_containerpushrepository")
+
+
+class ContainerQuerySetViewSet(viewsets.GenericViewSet):
+    def get_queryset(self):
+        qs = super().get_queryset()
+        has_model_push_repo = self.request.user.has_perm(push_repo.perm)
+        has_model_repo = self.request.user.has_perm(mirror_repo.perm)
+        if has_model_push_repo and has_model_repo:
+            return qs
+        # if has_model_repo:
+        #    return qs.model.objects.filter(repositories__pulp_type=mirror_repo.type)
+        # if has_model_push_repo:
+        #    return qs.model.objects.filter(repositories__pulp_type=push_repo.type)
+        query_params = self.request.query_params
+        if query_params and "repository_version" in query_params.keys():
+            policy = self.permission_classes[0]
+            repos_can_see, push_repos_can_see = policy.has_repo_query_params(
+                self, self.request, self, mirror_repo, push_repo
+            )
+        else:
+            content_type = qs.model.get_pulp_type()
+            repo_qs = models.ContainerRepository.objects.filter(
+                content__pulp_type=content_type, content__in=qs
+            ).only("pk")
+            push_repo_qs = models.ContainerPushRepository.objects.filter(
+                content__pulp_type=content_type, content__in=qs
+            ).only("pk")
+            from guardian.shortcuts import get_objects_for_user
+
+            push_repos_can_see = get_objects_for_user(
+                self.request.user, push_repo.perm, klass=push_repo_qs
+            ).values_list("pk")
+            repos_can_see = get_objects_for_user(
+                self.request.user, mirror_repo.perm, klass=repo_qs
+            ).values_list("pk")
+        content_qs = qs.model.objects.filter(repositories__in=repos_can_see)
+        push_content_qs = qs.model.objects.filter(repositories__in=push_repos_can_see)
```

Comment:
Wondering, if we can make that one filter:
```suggestion
        result_qs = qs.model.objects.filter(repositories__in=repos_can_see + push_repos_can_see)
```

Also by referring to `qs.model.objects` we are resetting every query that was implemented by `super().get_queryset()` before.